src/yunohost/app.py

```diff
@@ -65,66 +69,122 @@ def app_listlists():
+
+    # If a url and and a name is given, try to register new list,
+    # the fetch only this list
+    if url is not None:
+        if name:
+            _register_new_applist(url, name)
+            # Refresh the applists dict
+            applists = _read_applist_list()
+            applists_to_be_fetched = [name]
+        else:
+            raise MoulinetteError(errno.EINVAL,
+                                  m18n.n('custom_appslist_name_required'))
 
-    # Download file
-    try:
-        applist_request = requests.get(url, timeout=30)
-    except Exception as e:
-        raise MoulinetteError(errno.EBADR, m18n.n('appslist_retrieve_error', error=str(e)))
+    # If a name is given, look for an applist with that name and fetch it
+    elif name is not None:
+        if name not in applists.keys():
+            raise MoulinetteError(errno.EINVAL,
+                                  m18n.n('appslist_unknown', name=name))
+        else:
+            applists_to_be_fetched = [name]
 
-    if (applist_request.status_code != 200):
-        raise MoulinetteError(errno.EBADR, m18n.n('appslist_retrieve_error', error="404, not found"))
+    # Otherwise, fetch all lists
+    else:
+        applists_to_be_fetched = applists.keys()
 
-    # Validate app list format
-    # TODO / Possible improvement : better validation for app list (check that
-    # json fields actually look like an app list and not any json file)
-    applist = applist_request.text
-    try:
-        json.loads(applist)
-    except ValueError, e:
-        raise MoulinetteError(errno.EBADR, m18n.n('appslist_retrieve_bad_format'))
+    # Fetch all applists to be fetched
+    for name in applists_to_be_fetched:
 
-    # Write app list to file
-    list_file = '%s/%s.json' % (REPO_PATH, name)
-    with open(list_file, "w") as f:
-        f.write(applist)
+        url = applists[name]["url"]
 
-    # Setup a cron job to re-fetch the list at midnight
-    open("/etc/cron.d/yunohost-applist-%s" % name, "w").write('00 00 * * * root yunohost app fetchlist -u %s -n %s > /dev/null 2>&1\n' % (url, name))
+        logger.debug("Attempting to fetch list %s at %s" % (name, url))
 
-    logger.success(m18n.n('appslist_fetched'))
+        # Download file
+        try:
+            applist_request = requests.get(url, timeout=30)
+        except requests.exceptions.SSLError:
+            logger.error(m18n.n('appslist_retrieve_error',
+                                applist=name,
+                                error="SSL connection error"))
+            continue
+        except Exception as e:
+            logger.error(m18n.n('appslist_retrieve_error',
+                                applist=name,
+                                error=str(e)))
+            continue
+        if applist_request.status_code != 200:
+            logger.error(m18n.n('appslist_retrieve_error',
+                                applist=name,
+                                error="404, not found"))
```

Comment:
It could be a 500 or 403.
By redirections are followed by request.get, so it couldn't be a 301.